src/app/templates/partners-promotion/index.tsx

```diff
@@ -0,0 +1,123 @@
+import React, { memo, MouseEventHandler, useCallback, useEffect, useState } from 'react';
+
+import clsx from 'clsx';
+import { useDispatch } from 'react-redux';
+
+import Spinner from 'app/atoms/Spinner/Spinner';
+import { useAppEnv } from 'app/env';
+import { hidePromotionAction, setLastReportedPageNameAction } from 'app/store/partners-promotion/actions';
+import {
+  useLastReportedPageNameSelector,
+  useShouldShowPartnersPromoSelector,
+  usePromotionHidingTimestampSelector
+} from 'app/store/partners-promotion/selectors';
+import { AnalyticsEventCategory, useAnalytics } from 'lib/analytics';
+import { AD_HIDING_TIMEOUT } from 'lib/constants';
+
+import { HypelabPromotion } from './components/hypelab-promotion';
+import { OptimalPromotion } from './components/optimal-promotion';
+import styles from './partners-promotion.module.css';
+import { PartnersPromotionVariant } from './types';
+
+export { PartnersPromotionVariant } from './types';
+
+interface PartnersPromotionProps {
+  variant: PartnersPromotionVariant;
+  /** For distinguishing the ads that should be hidden temporarily */
+  id: string;
+  pageName: string;
+}
+
+const shouldBeHiddenTemporarily = (hiddenAt: number) => {
+  return Date.now() - hiddenAt < AD_HIDING_TIMEOUT;
+};
+
+export const PartnersPromotion = memo<PartnersPromotionProps>(({ variant, id, pageName }) => {
+  const isImageAd = variant === PartnersPromotionVariant.Image;
+  const { trackEvent } = useAnalytics();
+  const { popup } = useAppEnv();
+  const dispatch = useDispatch();
+  const hiddenAt = usePromotionHidingTimestampSelector(id);
+  const shouldShowPartnersPromo = useShouldShowPartnersPromoSelector();
+  const lastReportedPageName = useLastReportedPageNameSelector();
```

Comment:
One more problem about storing this value just in browser storage is that it should be reset on page reload to send "Internal Ads Activity" again. It's easy in the current implementation but too difficult with `useStorage`.

Reply:
It wouldn't work because `lastReportedPageName` will persist after the web extension reload, which is harmful in our case. It doesn't happen in the current implementation.